apache-maven/src/assembly/maven/conf/settings.xml

```diff
@@ -244,6 +244,35 @@ under the License.
       </properties>
     </profile>
     -->
+    <profile>
+      <activation>
+        <activeByDefault>true</activeByDefault>
```

Comment:
Here's the real issue. This means: activate if no other profiles are being activated. So if I run `mvn -P my-profile`, this one will be disabled, meaning we won't have remote repoitories.

The proper way to activate is by using `<activeProfiles>`. Again a sign to me we need to change the settings.xml

Reply:
I was never a fan of implicit deactivation because it seems to much magic to me. My initial idea was to name the profile. Same approach I have used with Maven Resolver Redisson Sync Context. I want to reserve a URN-style id namespace to use. In Resolver I have used: `maven:resolver:` prefix. Here I would like to use `maven:core:central-repo`, at least the `maven:core:` prefix.

Reply:
Officially `:` is not a valid character for ids, but I'm not sure if this was ever validated.

Reply:
No, it is not. Colon is invalid for group/artifact ids. For element ids nothing is enforced. See: https://issues.apache.org/jira/browse/MNG-7052. I intionally did not want to use a period because people might have used `maven.repo...`.

Reply:
We also do not have an official documentation what is valid or not.

Reply:
The implementation of ModelValidator is quite clear about the definition of `id`, but former Maven committers forget to do a strict validation for the complete model.

Reply:
The `DefaultModelValidator` retains a negative list of characters, not a positive one. Are you referring to another validation? I am confused.

Reply:
Added a profile. Have a look.